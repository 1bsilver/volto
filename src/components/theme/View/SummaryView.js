/**
 * Summary view component.
 * @module components/theme/View/SummaryView
 */

import React from 'react';
import Helmet from 'react-helmet';
import { Link } from 'react-router';

/**
 * Summary view component class.
 * @function SummaryView
 * @params {object} content Content object.
 * @returns {string} Markup of the component.
 */
const SummaryView = ({ content }) => (
  <div id="page-home">
    <Helmet title={content.title} />
    <div className="container">
      <article id="content">
        <header>
          <h1 className="documentFirstHeading">{content.title}</h1>
          {content.description && <div className="documentDescription description">{content.description}</div>}
        </header>
        <section id="content-core">
          {content.items.map(item =>
            <article className="tileItem"
                     key={item.url}>
              <h2 className="tileHeadline">
                <Link to={item.url}
                      className="summary url"
                      title={item['@type']}>
                  {item.title}
                </Link>
              </h2>
              {item.description &&
                <div className="tileBody">
                  <span className="description">{item.description}</span>
                </div>
              }
              <div className="tileFooter">
                <Link to={item.url}>
                  Read Moreâ€¦
                </Link>
              </div>
              <div className="visualClear"> </div>
            </article>
          )}
        </section>
       </article>
    </div>
  </div>
);

export default SummaryView;
