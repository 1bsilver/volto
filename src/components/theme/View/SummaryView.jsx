/**
 * Summary view component.
 * @module components/theme/View/SummaryView
 */

import React, { PropTypes } from 'react';
import Helmet from 'react-helmet';
import { Link } from 'react-router';

/**
 * Summary view component class.
 * @function SummaryView
 * @params {object} content Content object.
 * @returns {string} Markup of the component.
 */
const SummaryView = ({ content }) => (
  <div id="page-home">
    <Helmet title={content.title} />
    <div className="container">
      <article id="content">
        <header>
          <h1 className="documentFirstHeading">{content.title}</h1>
          {content.description && <div className="documentDescription description">{content.description}</div>}
        </header>
        <section id="content-core">
          {content.items.map(item =>
            <article
              className="tileItem"
              key={item.url}
            >
              <h2 className="tileHeadline">
                <Link
                  to={item.url}
                  className="summary url"
                  title={item['@type']}
                >
                  {item.title}
                </Link>
              </h2>
              {item.description &&
                <div className="tileBody">
                  <span className="description">{item.description}</span>
                </div>
              }
              <div className="tileFooter">
                <Link to={item.url}>
                  Read Moreâ€¦
                </Link>
              </div>
              <div className="visualClear" />
            </article>,
          )}
        </section>
      </article>
    </div>
  </div>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
SummaryView.propTypes = {
  content: PropTypes.shape({
    title: PropTypes.string,
    description: PropTypes.string,
    items: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string,
        description: PropTypes.string,
        url: PropTypes.string,
        '@type': PropTypes.string,
      }),
    ),
  }).isRequired,
};

export default SummaryView;
