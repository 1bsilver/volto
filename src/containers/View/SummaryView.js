/**
 * Summary view component.
 * @module containers/SummaryView
 */

import React from 'react';
import Helmet from 'react-helmet';
import { Link } from 'react-router';

/**
 * Summary view component class.
 * @function SummaryView
 * @params {object} page Page object.
 * @returns {string} Markup of the component.
 */
const SummaryView = ({ page }) => (
  <div id="page-home">
    <Helmet title={page.title} />
    <div className="container">
      <article id="content">
        <header>
          <h1 className="documentFirstHeading">{page.title}</h1>
          {page.description && <div className="documentDescription description">{page.description}</div>}
        </header>
        <section id="content-core">
          {page.items.map(item =>
            <article className="tileItem"
                     key={item.url}>
              <h2 className="tileHeadline">
                <Link to={item.url}
                      className="summary url"
                      title={item['@type']}>
                  {item.title}
                </Link>
              </h2>
              {item.description &&
                <div className="tileBody">
                  <span className="description">{item.description}</span>
                </div>
              }
              <div className="tileFooter">
                <Link to={item.url}>
                  Read Moreâ€¦
                </Link>
              </div>
              <div className="visualClear"> </div>
            </article>
          )}
        </section>
       </article>
    </div>
  </div>
);

export default SummaryView;
