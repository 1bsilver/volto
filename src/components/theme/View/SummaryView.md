SummaryView example source:

```jsx_ noeditor
const { Provider } = require('react-intl-redux');
const configureStore = require('redux-mock-store').default;
import SummaryView from './SummaryView';
import {StaticRouter} from 'react-router-dom';
const store = configureStore()({
  userSession: {
    login: {},
  },
  intl: {
    locale: 'en',
    messages: {},
  },
  users: {
    reset: {
      loading: '',
    },
  },
  actions: {},
  content: { get: {} },
});

<div className={'rsg--pre-42'}>
  <Provider store={store}>
    <StaticRouter>
      <SummaryView
        content={{
          title: 'plone',
          description: 'open source CMS',
          items: [
            {
              '@id': '@plone',
              '@type': '@image',
              url: 'www.plone.org',
              title: 'plone',
              description: 'open source CMS',
              image: { scales: { thumb: {} } },
              image_caption: '',
            },
          ],
        }}
      />
    </StaticRouter>
  </Provider>
</div>;
```
