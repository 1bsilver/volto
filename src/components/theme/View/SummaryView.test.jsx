import React from 'react';
import renderer from 'react-test-renderer';
import SummaryView from './SummaryView';

test('renders a summary view component', () => {
  const component = renderer.create(
    <SummaryView
      content={{
        title: 'Hello World!',
        description: 'Hi',
        items: [{
          title: 'My item',
          description: 'My item description',
          url: 'http://item',
          '@type': 'Document',
        }],
      }}
    />
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
