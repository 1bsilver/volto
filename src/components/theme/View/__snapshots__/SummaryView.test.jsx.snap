// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a summary view component 1`] = `
<div
  id="page-home"
>
  <article
    id="content"
  >
    <header>
      <h1
        className="documentFirstHeading"
      >
        Hello World!
      </h1>
      <p
        className="description"
      >
        Hi
      </p>
    </header>
    <section
      id="content-core"
    >
      <article>
        <h2>
          <a
            onClick={[Function]}
            style={Object {}}
            title="Document"
          >
            My item
          </a>
        </h2>
        <p>
          My item description
        </p>
        <p>
          <a
            onClick={[Function]}
            style={Object {}}
          >
            Read Moreâ€¦
          </a>
        </p>
      </article>
    </section>
  </article>
</div>
`;
